### ISSUE TITLE
Midi Learn on VST's

### ISSUE THREAD

**Issue by Kimaker** [raphaelquast/beatstep#82]
Hello community. A big congratulations and a big thank you Raphael for your work and your script which transforms the Beatstep into a real versatile controller.

However, I am unable to map encoders 1-4 & 9-12 on VSTs like Kontakt or Analog Lab, whereas with Ableton's effect or instrument macros, the mapping is done automatically and perfectly.

It's as if they were not recognized by the VST, is this normal?

Thanks in advance

**Comment by markusschloesser** [comment]
Not Raphael, but also a dev for a Midi Remote Script (The Mackie C4).
Not all plug-ins expose their parameters initially. Here's how to do that manually: [Ableton - expose plug-in parameters](https://help.ableton.com/hc/en-us/articles/209073089-Saving-plug-in-parameter-configurations). For m4l devices it is a bit more complicated. For that you need to open the device in maxforlive, go to "Parameters" and set the visibility of the parameter you want to "visible and automatable (or something like this) and then store the device. Be careful to not include all parameters as that will spam your undo history.

**Comment by raphaelquast** [comment]
@markusschloesser thanks a lot for responding! (I have little experience with the subtleties of VSTs or Max so your insights are much appreciated)

**Comment by Kimaker** [comment]
Yes thank you very much @markusschloesser for taking the time to provide this very informative answer and and a solution that I no longer hoped for. It's so great, thanks to your technique I was able to correctly map the controllers. It's a kind of Midi Learn managed by Ableton, right?
Is it possible to find/save these "Configure Mode" mappings in order to import them on the Ableton from another machine?

Two other questions more general question about the script:
- Is there a Control mode that would allow to use/assign all the controllers (pads & encoders) and not the first twelve? 

- I would like to assign the Transpose controller to a Volume parameter of a device, but it doesn't seem accessible. Can it be?

Thank you again

**Comment by markusschloesser** [comment]
> Is it possible to find/save these "Configure Mode" mappings in order to import them on the Ableton from another machine?

you could copy the Live preference and/or library file, but that of course brings more than just the plugin config.
